platform/linux-generic/odp_packet.c

```diff
@@ -1286,7 +1286,7 @@ int odp_packet_l2_offset_set(odp_packet_t pkt, uint32_t offset)
 {
 	odp_packet_hdr_t *pkt_hdr = packet_hdr(pkt);
 
-	if (offset >= pkt_hdr->frame_len)
+	if (offset > pkt_hdr->frame_len)
```

Comment:
This change requires an API change as the spec says relevant offsets must be in the range `0..odp_packet_len(pkt) - 1` .  Same comment for the L3 and L4 changes in this patch.

In theory the validation tests should test these bounds, but as with most parts of the API violations simply result in undefined behavior, so this is an "honor system". Still, we can't violate the spec here without changing the spec.

Reply:
Hmm. We've spent several minutes on this, but nobody reminded of API convention. Should we change the spec here? @psavol what is your opinion?